Tools/android/GenerateAndroidMk.py

```diff
@@ -0,0 +1,224 @@
+        tpl = tpl.replace("@LOCAL_CFLAGS", self.getDefines("CXX_DEFINES"))
+        tpl = tpl.replace("@LOCAL_C_INCLUDES", self.getIncludes())
+        with open(path.join(self.src_, "Android.mk"), "w") as f:
+            f.write(tpl)
+
+    #virtuall functions
+    def getTemplatePath(self):
+        raise Exception("pure virtul function")
+
+    def getMakefile(self):
+        raise Exception("pure virtul function")
+
+    def getFlagsfile(self):
+        raise Exception("pure virtul function")
+
+    def adjustSources(self, lines):
+        print(lines)
+        return lines
+
+    def adjustIncludes(self, lines):
+        print(lines)
+        return lines
+
+    def getCmakeCmd(self):
+        return "cmake " + self.src_
+
+    def generateMakefile(self):
+        #windows can help us debug the script
+        #but we do not want generate makefile on widnows
+        if os.name == "nt":
+            return
+        cmd = "rm " + path.join(self.build_, "CMakeCache.txt") + "> /dev/null 2>&1;"
+        cmd += "mkdir " + self.build_ + "> /dev/null 2>&1;"
+        cmd += "cd " + self.build_
+        cmd += "&& " + self.getCmakeCmd() + "> /dev/null 2>&1"
+        os.system(cmd)
+
+    def getIncludes(self):
+        text = self.getDefines("CXX_INCLUDES")
+        includes = []
+        lines = text.split("\n")
+        for l in lines:
+            #normpath will make sure we did not refer outside.
+            p = path.normpath(l)
+            j = p.find(self.src_)
+            if j != -1:
+                includes.append(p[j:].replace(self.src_, "$(LOCAL_PATH)"))
+        return INDENT + ("\n" + INDENT).join(includes)
+
+    def getDefines(self, name):
+        flags = self.getFlagsfile()
+        with open(flags) as f:
+            text = f.read()
+        line = re.findall(name + " =.*\n", text)[0]
+        line = line.replace(name + " = ", "")
+        line = INDENT + line.replace(" ", " \\\n" + INDENT)
+        return line
+
+    def getSources(self):
+        makefile = self.getMakefile()
+        with open(makefile) as f:
+            text = f.read()
+        lines = re.findall(".*?\\.o:\\n", text)
+        lines = [l.replace(".o:\n", " \\") for l in lines]
+        self.adjustSources(lines)
+        #make source pretty
+        return INDENT + ("\n" + INDENT).join(lines)
+
+
+class GmmGeneator(Generator):
+    def __init__(self, root):
+        src = path.join(root, "gmmlib")
+        templ = path.join(src,  "Tools/android")
+        super(GmmGeneator, self).__init__(src, templ)
+
+    def getTemplatePath(self):
+        return path.join(self.templ_, "gmm.Android.mk.template")
+
+    def getMakefile(self):
+        return path.join(self.build_, "Source/GmmLib/Makefile")
+
+    def getFlagsfile(self):
+        return path.join(self.build_, "Source/GmmLib/CMakeFiles/gmm_umd.dir/flags.make")
+
+    def adjustSources(self, lines):
+        for i, l in enumerate(lines):
+            j = l.find("__/")
```

Comment:
Currently, if you use `grep -ir \\.o:$ Makefile  |sed s/.c.o:/.c/ |sed s/.cpp.o:/.cpp/ | sed s,^,\\t, | sed s/$/' \\'/` to grep makefile in gmmlib, you will see __/xxxx, could we conside adjust the cmake to remove __/?

Reply:
could you elaborate the problem, such like pasting some log?
Do you have suggestion what should be changed in cmake and which side media or gmmlib should do this change? Thanks

Reply:
attached gmmlib and driver make file
if you grep them using 
"grep -ir \\.o:$ Makefile |sed s/.c.o:/.c/ |sed s/.cpp.o:/.cpp/ | sed s,^,\\t, | sed s/$/' \\'/"
you will find the gmmlib output like this

```
        Utility/GmmUtility.cpp \
        __/Common/AssertTracer/AssertTracer.cpp \
```
it has two issues I hope we can  fix.
1. the __ prefix in __/Common/AssertTracer/AssertTracer.cpp
2. Utility/GmmUtility.cpp  not use the root directory, we need append "Source/GmmLib" before it

If you grep the driver makefile. all things is perfect like this. 

```
        linux/gen10_cnl/ddi/media_libva_caps_g10_cnl.cpp \
        linux/gen8/ddi/media_libva_caps_g8.cpp \
        linux/gen8/ddi/media_sku_wa_g8.cpp \
        linux/gen8/ddi/media_sysinfo_g8.cpp \
        linux/gen9/ddi/media_libva_caps_g9.cpp \
```

[Makefile.gmmlib.txt](https://github.com/intel/media-driver/files/1969560/Makefile.gmmlib.txt)

[Makefile.driver.txt](https://github.com/intel/media-driver/files/1969556/Makefile.driver.txt)